theano/sandbox/cuda/cuda_ndarray.cu

```diff
@@ -1030,7 +1030,6 @@ CudaNdarray_TakeFrom(CudaNdarray * self, PyObject *args){
         PyObject * indices_float32 = NULL;
         indices_float32 = PyArray_View((PyArrayObject*)indices_obj,
                                                   float32_descr, NULL);
-        Py_DECREF(float32_descr);
```

Comment:
NumPy C-API [says](http://docs.scipy.org/doc/numpy/reference/c-api.array.html#data-type-descriptors):
> Functions that take PyArray_Descr * objects and return arrays steal references to the data-type their inputs unless otherwise noted. Therefore, you must own a reference to any data-type object used as input to such a function.

So I'm pretty sure we should not decref it after `PyArray_View` is called.
I tried to run Caglar's code with this PR and it seemed to work, without any sign of memory leak. I did not try without the C code at all.

I can also fix line 1068.